### ISSUE TITLE
Validate Web3 Secret Storage's UUID consistency

### ISSUE THREAD

**Issue by BasixKOR** [planetarium/libplanet#2314]
When working with Web3 Secret Storage, it is possible to create a malformed key file which has a different `id` field from the filename. This may cause malfunction as some codes read UUID from filenames while some reads from the `id` field. We need to validate the key file, so the UUIDs are always consistent.

**Comment by Akamig** [comment]
I've checked the implementation of Web3KeyStore and found out that the GUID of the filename is always prioritized over in-file Guid.

Currently, `ProtectedPrivateKey Get(Guid id)` and `ProtectedPrivateKey Get(string name)` is checking given GUID using `ListFiles()`, which is only returning `IEnumerable<(Guid, string)>` with matching pattern path (UTC something).

**Comment by BasixKOR** [comment]
Does it still apply if it is accessed through `ProtectedPrivateKey` retrieved from `Get`?

**Comment by Akamig** [comment]
If you do Get(Guid A) and there is file with the name GUID A, but the ID of the contents of the file is GUID B, Returned ProtectedPrivateKey will have GUID B.

**Comment by Akamig** [comment]
Oh, I'm sorry, I confusingly used 'prioritized'. 
What I'm trying to say is, 

- **`ProtectedPrivateKey` doesn't have `id` property.** 
- `id` solely exists for distinguishing specific JSON, 
- and no function is **Reading** `id` in JSON to validate if matching with filename.
    - `ProtectedPrivateKey.FromJson()` not fetches `id`
    - `ProtectedPrivateKey Get(string name)` don't validate id, simply call FromJson().
    - `ProtectedPrivateKey Get(Guid id)` in Web3KeyStore just uses `ListFiles()`
    - `IEnumerable<(Guid, string)> ListFiles()` in Web3KeyStore reads filename only.
- In conclusion. `id` in JSON **only gets written, never read.**

My suggestion is, as id never mismatches while writing a new Web3Key, and although PPK is validating JSON in FromJson, id property is only for Web3KeyStore, let's just add checking logic in `Get(Guid id)`.

...but on my second opinion, it seems plausible to add this validation logic on PPK side too. since we can't validate with the first suggestion when `Get(string name)` or `PPK.FromJson()` is used directly.

What do you think?

**Comment by BasixKOR** [comment]
As `id` field isn't actually used I think it is safe to close this issue as the inconsistency won't cause any unexpected behaviour.
